packages/internal/src/utils.js

```diff
@@ -266,11 +266,11 @@ harden(bindAllMethods);
  * A more constrained version of {deeplyFulfilled} for type safety until https://github.com/endojs/endo/issues/1257
  * Useful in starting contracts that need all terms to be fulfilled in order to be durable.
  *
- * @type {<T extends {}>(unfulfilledTerms: T) => import('@endo/eventual-send').ERef<DeeplyAwaited<T>>}
+ * @type {<T extends {}>(unfulfilledTerms: T) => Promise<DeeplyAwaited<T>>}
  */
 export const deeplyFulfilledObject = obj => {
   assert(isObject(obj), 'param must be an object');
```

Comment:
This will throw synchronously rather than returning a rejected promise, which seems to raise the same concerns as...

> Using `await` inside a conditional means that _sometimes_ the function is split into two pieces, and sometimes it is not, which makes this review process much harder.
 -- https://github.com/Agoric/agoric-sdk/wiki/No-Nested-Await

Reply:
`deeplyFulfilled` seems to exhibit the same hazard:

```
  if (!isObject(val)) {
    return val;
  }
```

I wonder why that's OK.

Reply:
> I wonder why that's OK.

Since `deeplyFulfilledObject` and `deeplyFulfilled` are not `async` functions, they don't violate the `await` rules (i.e. they have no `await`s).  They always return synchronously in the same turn.  Even if that synchronous return value is sometimes a Promise.